test/unit/local/steps/add_infos.js

```diff
@@ -0,0 +1,75 @@
+/* eslint-env mocha */
+/* @flow */
+
+/*::
+import type { AtomWatcherEvent } from '../../../../core/local/steps/event'
+*/
+
+const should = require('should')
+const addInfos = require('../../../../core/local/steps/add_infos')
+const Buffer = require('../../../../core/local/steps/buffer')
+
+describe('core/local/steps/add_infos', () => {
+  it('should returns an enhanced batch with infos', async () => {
+    const batch = [
+      {
+        action: 'scan',
+        kind: 'unknown',
+        path: ''
+      }
+    ]
+    const buffer = new Buffer()
+    buffer.push(batch)
+    const enhancedBuffer = addInfos(buffer, {
+      syncPath: ''
+    })
+    const enhancedBatch = await enhancedBuffer.pop()
+    should(enhancedBatch)
+      .be.an.Array()
+      .and.have.length(batch.length)
+  })
+
+  it('should add specific infos for specific events', async () => {
+    const batch = [
+      {
+        action: 'scan',
+        kind: 'unknown',
+        path: ''
+      },
+      {
+        action: 'created',
+        kind: 'unknown',
+        path: ''
+      },
+      {
+        action: 'modified',
+        kind: 'unknown',
+        path: ''
+      },
+      {
+        action: 'renamed',
+        kind: 'unknown',
+        path: ''
+      }
+    ]
+    const buffer = new Buffer()
+    buffer.push(batch)
+    const enhancedBuffer = addInfos(buffer, {
+      syncPath: ''
+    })
+    const [scanEvent, ...otherEvents] = await enhancedBuffer.pop()
+    should(scanEvent).eql({
+      action: 'scan',
+      kind: 'file',
+      path: '',
+      _id: ''
```

Comment:
Maybe you can use something else than an empty string for `path` to check if the `_id` is correctly built

Reply:
Oh, I did not notice that the ID could be built from the path.
I'll do it, thanks for the tips.

Reply:
I have a builder ready in my branch. I can push that first so it can be reused right away?

Reply:
With flow I don't see the point to use
```
const event = new AtomWatcherEvent('scan', 'file', __filename)
```
instead of an plain old objet
```
const event = { action: 'scan', kind: 'file', path: __filename }
```